src/python/pants/backend/python/tasks/setup_py.py

```diff
@@ -597,13 +598,19 @@ def create_setup_py(self, target, dist_dir):
     shutil.move(chroot.path(), setup_dir)
     return setup_dir, reduced_deps
 
+  def _get_exported_targets(self):
+    if self.context.products.is_required_data(self.PYTHON_DISTS_RECURSIVE_PRODUCT):
```

Comment:
Note that you'll need to actually populate this new product when it has been requested, as is kicked off on this line for the existing product: https://github.com/pantsbuild/pants/blob/4be101000cb9179a9ee5edd1fc7e74af3a00348d/src/python/pants/backend/python/tasks/setup_py.py#L633